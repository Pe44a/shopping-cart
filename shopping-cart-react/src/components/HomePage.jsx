import NavBar from "./NavBar"
import { Link } from "react-router-dom";

function HomePage() {

  return (
    <>
      <NavBar/>

      <div className='main'>
        <Link className="shopLink" to="/shop">Shop now</Link>
      </div>
    </>
  )
}

export default HomePage