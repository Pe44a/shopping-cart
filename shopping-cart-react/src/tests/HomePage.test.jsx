import { render, screen } from "@testing-library/react";
import { describe, it, expect } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import userEvent from "@testing-library/user-event";
import HomePage from "../components/HomePage";


it("Home page snapshot", async() => {

    const { container } = render(<>
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    </>);

    await expect(container).toMatchSnapshot();
})