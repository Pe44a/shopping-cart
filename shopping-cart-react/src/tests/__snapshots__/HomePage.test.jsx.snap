// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page snapshot 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="shop-name"
    >
      Some-shop
    </div>
    <div
      class="nav-bar"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Shop
      </a>
      <a
        href="/cart"
      >
        Cart
      </a>
    </div>
  </div>
  <div
    class="main"
  />
</div>
`;
